src/NServiceBus.Core/Recoverability/Faults/MoveFaultsToErrorQueueBehavior.cs

```diff
@@ -31,39 +53,36 @@ public MoveFaultsToErrorQueueBehavior(CriticalError criticalError, IPipelineBase
             {
                 throw;
             }
-            catch (Exception exception)
+            catch (Exception e)
             {
-                try
-                {
-                    var message = context.Message;
+                faultsStatusStorage.AddException(uniqueMessageId, e);
```

Comment:
So essentially your storage could store the info instead of the exception directly